q2_cutadapt/tests/test_demux.py

```diff
@@ -163,6 +163,43 @@ def test_variable_length_barcodes(self):
         self.assert_demux_results(metadata.to_series(), obs_demuxed_art)
         self.assert_untrimmed_results(b'', obs_untrimmed_art)
 
+    def test_batch_size(self):
+        metadata = CategoricalMetadataColumn(
+            pd.Series(['AAAA', 'CCCC'], name='Barcode',
+                      index=pd.Index(['sample_a', 'sample_b'], name='id')))
+
+        with redirected_stdio(stderr=os.devnull):
+            obs_demuxed_art, obs_untrimmed_art = \
+                self.demux_single_fn(self.muxed_sequences, metadata,
+                                     batch_size=1)
+
+        self.assert_demux_results(metadata.to_series(), obs_demuxed_art)
+        self.assert_untrimmed_results(b'@id6\nGGGGACGTACGT\n+\nzzzzzzzzzzzz\n',
+                                      obs_untrimmed_art)
+
+    def test_invalid_batch_size(self):
+        metadata = CategoricalMetadataColumn(
+            pd.Series(['AAAA', 'CCCC'], name='Barcode',
+                      index=pd.Index(['sample_a', 'sample_b'], name='id')))
+
+        with self.assertRaisesRegex(ValueError, '2.*received 5'):
+            self.demux_single_fn(self.muxed_sequences, metadata, batch_size=5)
+
```

Comment:
Is it worth including a test for batch_size out-of-range. It would guard against removal of that Range predicate, but doesn't really add much of substance.
```suggestion
with self.assertRaisesRegex(TypeError,
                            r'batch_size.*-9999.*Range\(0, None\)'):
    self.demux_single_fn(self.muxed_sequences,
                         metadata, batch_size=-9999)
```

Reply:
I'm okay with not testing that, since the framework is responsible for ensuring the `Range` works as advertised, but I am glad you're thinking of these things - thanks!